particles/MeshParticles.cpp

```diff
@@ -23,11 +26,12 @@ TexturedMeshForParticles::TexturedMeshForParticles()
     vertexCount = myData.vertices_length / 3;
 }
 
-const TexturedMeshForParticles& TexturedMeshForParticles::instance() noexcept
+const TexturedMeshForParticles& TexturedMeshForParticles::instance(std::string meshName) noexcept
 {
-    static const TexturedMeshForParticles newmesh;
+    static const TexturedMeshForParticles newmesh(meshName);
```

Comment:
okay, this is why this does not work :) 

here it was a good way as long as we were having a single mesh as it creates a static (singleton).
Now instead we don't want to have an instance() method but create a new TexturedMeshForParticles object every time our mesh changes. Thus a TexturedMeshForParticles should be stored in the RenderedNode class